pyro/ops/stats.py

```diff
@@ -350,57 +349,46 @@ def fit_generalized_pareto(X):
     """
     Given a dataset X assumed to be drawn from the Generalized Pareto
     Distribution, estimate the distributional parameters k, sigma using a
-    variant of the technique described in reference [1]. This
-    implementation is modified from the implementation used in reference [2].
+    variant of the technique described in reference [1], as described in
+    reference [2].
 
     References
     [1] 'A new and efficient estimation method for the generalized Pareto distribution.'
     Zhang, J. and Stephens, M.A. (2009).
     [2] 'Pareto Smoothed Importance Sampling.'
     Aki Vehtari, Andrew Gelman, Jonah Gabry
-    https://github.com/avehtari/PSIS/blob/904146236767182270c9718a7b2a30831fe701fe/py/psis.py#L211
 
-    :param torch.Tensor or numpy.ndarray X: the input data X
+    :param torch.Tensor: the input data X
     :returns tuple: tuple of floats (k, sigma) corresponding to the fit parameters
     """
-    if isinstance(X, torch.Tensor):
-        X = X.data.cpu().float().numpy()
+    if not isinstance(X, torch.Tensor) or X.dim() != 1:
+        raise ValueError("Input X must be a 1-dimensional torch tensor")
 
-    if X.ndim != 1 or len(X) <= 1:
-        raise ValueError("Invalid input array.")
+    X = X.cpu().double()
+    X = torch.sort(X, descending=False)[0]
```

Comment:
is the conversion to cpu because `sort()` is only cpu implemented?

Reply:
cuz i want doubles

Reply:
there are double tensors on cuda though?